app/src/main/java/app/revanced/manager/ui/viewmodel/MainViewModel.kt

```diff
@@ -30,4 +41,70 @@ class MainViewModel(
             }
         }
     }
-}
\ No newline at end of file
+
+    fun launchLegacyActivity(): Boolean {
+        return try {
+            val intent = Intent()
+            intent.setClassName(
+                "app.revanced.manager.flutter",
+                "app.revanced.manager.flutter.ExportSettingsActivity"
+            )
+            launcher.launch(intent)
+            true
+        } catch (e: Exception) {
+            false
+        }
+    }
+
+    fun applyLegacySettings(data: String) = viewModelScope.launch {
+        val json = Json { ignoreUnknownKeys = true }
+        val settings = json.decodeFromString<LegacySettings>(data)
+
+        settings.themeMode?.let { theme ->
+            val themeMap = mapOf(
+                0 to Theme.SYSTEM,
+                1 to Theme.LIGHT,
+                2 to Theme.DARK
+            )
+            prefs.theme.update(themeMap[theme]!!)
+        }
+        settings.useDynamicTheme?.let { dynamicColor ->
+            prefs.dynamicColor.update(dynamicColor == 1)
+        }
+        settings.apiUrl?.let { api ->
+            prefs.api.update(api.removeSuffix("/"))
+        }
+        settings.experimentalPatchesEnabled?.let { allowExperimental ->
+            prefs.allowExperimental.update(allowExperimental == 1)
+        }
+        settings.patchesAutoUpdate?.let { autoUpdate ->
+            with(patchBundleRepository) {
+                sources
+                    .first()
+                    .find { it.uid == 0 }
+                    ?.asRemoteOrNull
+                    ?.setAutoUpdate(autoUpdate == 1)
+
+                updateCheck()
+            }
+        }
+        settings.patchesChangeEnabled?.let {
+            // TODO: Implement setting
+        }
+        settings.showPatchesChangeWarning?.let {
+            // TODO: Implement setting
+        }
```

Comment:
The selection warning was recently implemented. Can you implement this?

Reply:
How do I get the latest changes from Compose dev onto this branch?

Reply:
git fetch
git merge compose-dev

or 

git fetch
git rebase compose-dev

Reply:
Merging like osu said is the easiest way. You might have to specify `origin/compose-dev` instead of `compose-dev` though

Reply:
Should I use the `patchesChangeEnabled` or the `showPatchesChangeWarning` setting from Flutter to control the Compose preference.